### DISCUSSION TITLE
Nova::withBreadcrumbs is only respected on first request, not subsequent inertia requests

### DISCUSSION THREAD

**Discussion by puzzledmonkey** [laravel/nova-issues#5155] [Ideas & Feature Requests]
- Laravel Version: 9.41.0
- Nova Version: 4.19.4
- PHP Version: 8.1.12
- Database Driver & Version: n/a
- Operating System and Version: n/a
- Browser type and version: n/a
- Reproduction Repository: n/a

### Description:

When using a Nova::withBreadcrumbs callback that might change per request, it appears this is only being evaluated on the first page load and afterwards is ignored. For example, checking for LensRequests and disabling breadcrumbs on all lens pages doesn't work.

### Detailed steps to reproduce the issue on a fresh Nova installation:

Create a resource with a lens
Add this code to NovaServiceProvider boot()

```
Nova::withBreadcrumbs(function (NovaRequest $request) {
    return !($request instanceof LensRequest);
});
```

Observe that visiting the lens page normally via an inertia request does NOT hide the breadcrumb bar. A page reload (on the lens page) will hide it. However, once hidden, clicking back onto a normal resource page (via an inertia request) will NOT show the breadcrumb bar until you once again reload the page on a normal resource page.

It seems that the withBreadcrumbs setting is only being used from the initial page load, and all further inertia requests don't change the visibility of the breadcrumb bar.

**Comment by crynobone** [comment]
This is the current expected behaviour and was already considered during the development. Marking as feature request and will reconsider if there huge demand for it.

**Reply by puzzledmonkey** [reply]
@crynobone ok thanks for the clarification - is there any way we can modify the breadcrumbs returned from resource/lens classes?

**Comment by puzzledmonkey** [comment]
Closing due to apathy
